src/test/java/api/exercise/Exercise2.java

```diff
@@ -0,0 +1,123 @@
+package api.exercise;
+
+import org.hamcrest.Matchers;
+import org.junit.jupiter.api.Test;
+
+import java.util.Arrays;
+import java.util.function.BinaryOperator;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+@SuppressWarnings("unused")
+class Exercise2 {
+
+    @Test
+    void sequentialPrefixTestSum() {
+        Integer[] original = {1, 2, 3, 4, 5, 6, 7, 8};
+
+        Integer[] result = sequentialPrefix(original, Integer::sum);
+
+        assertThat(original, Matchers.not(Matchers.sameInstance(result)));
+        assertThat(result, Matchers.arrayContaining(1, 3, 6, 10, 15, 21, 28, 36));
+    }
+
+    @Test
+    void sequentialPrefixTestMult() {
+        Integer[] original = {1, 2, 3, 4, 5, 6, 7, 8};
+
+        Integer[] result = sequentialPrefix(original, Math::multiplyExact);
+
+        assertThat(original, Matchers.not(Matchers.sameInstance(result)));
+        assertThat(result, Matchers.arrayContaining(1, 2, 6, 24, 120, 720, 5_040, 40_320));
+    }
+
+    @Test
+    void sequentialPrefixTestLiningByMax() {
+        Integer[] original = {1, 2, 3, 2, 1, 5, 7, 5};
+
+        Integer[] result = sequentialPrefix(original, Integer::max);
+
+        assertThat(original, Matchers.not(Matchers.sameInstance(result)));
+        assertThat(result, Matchers.arrayContaining(1, 2, 3, 3, 3, 5, 7, 7));
+    }
+
+    /**
+     * Выполняет операцию сканирования в однопоточном режиме.
+     * Не модифицирует исходный набор данных.
+     * @param source Массив исходных элементов.
+     * @param operator Оператор сканирования.
+     * @return Результат сканирования.
+     * @see <a href="https://habr.com/company/epam_systems/blog/247805">Сканирование</a>
+     */
+    private static <T> T[] sequentialPrefix(T[] source, BinaryOperator<T> operator) {
+        T[] result = Arrays.copyOf(source, source.length);
+        Arrays.parallelPrefix(result, operator);
```

Comment:
I guess we should do this using the algorithm from the article

Reply:
No, I don't thinl so.